Framework/Beamline/inc/MantidBeamline/DetectorInfo.h

```diff
@@ -95,32 +94,27 @@ class MANTID_BEAMLINE_DLL DetectorInfo {
   void setRotation(const std::pair<size_t, size_t> &index,
                    const Eigen::Quaterniond &rotation);
 
-  size_t scanCount(const size_t index) const;
-  std::pair<int64_t, int64_t>
-  scanInterval(const std::pair<size_t, size_t> &index) const;
-  void setScanInterval(const size_t index,
-                       const std::pair<int64_t, int64_t> &interval);
+  size_t scanCount() const;
+  const std::vector<std::pair<int64_t, int64_t>> &scanIntervals() const;
   void setScanInterval(const std::pair<int64_t, int64_t> &interval);
 
-  void merge(const DetectorInfo &other);
   void setComponentInfo(ComponentInfo *componentInfo);
   bool hasComponentInfo() const;
   double l1() const;
   Eigen::Vector3d sourcePosition() const;
   Eigen::Vector3d samplePosition() const;
 
+  friend class ComponentInfo;
```

Comment:
Probably we should add a comment explaining why we need the friend (to use `merge`), and then also explain why merge is private. Otherwise no one will remember in a years time.